### PR TITLE
[FOR INFORMATION ONLY] Let's see what scalafmt can do

### PR DESCRIPTION
**Pull request by AWare**
Last time we looked at scalafmt, it crashed when we tried to run in on the whole codebase. 

It works now, here's another PR to discuss the benefits of autocode styling. 

This should eliminate random whitespace changes from intellij,

### CODE REVIEW

**Comment by joelochlann** [comment]
:-1: this is a great demonstration of why we won't be using `scalafmt` :stuck_out_tongue_winking_eye:

File: frontend/app/actions/ActionRefiners.scala [hunk 1]
```diff
@@ -22,33 +22,39 @@ import scala.concurrent.Future
...
   def resultModifier(f: Result => Result) = new ActionBuilder[Request] {
-    def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]) = block(request).map(f)
+    def invokeBlock[A](
+        request: Request[A], block: (Request[A]) => Future[Result]) =
+      block(request).map(f)
   }
 
-  def authenticated(onUnauthenticated: RequestHeader => Result = chooseRegister(_)): ActionBuilder[AuthRequest] =
-    new AuthenticatedBuilder(AuthenticationService.authenticatedIdUserProvider, onUnauthenticated)
+  def authenticated(onUnauthenticated: RequestHeader => Result = chooseRegister(
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: weird

File: frontend/app/actions/ActionRefiners.scala [hunk 2]
```diff
@@ -57,78 +63,119 @@ object ActionRefiners extends LazyLogging {
...
+              FreeSubscriber(_, member), PaidSubscriber(_, member))
+      }).run
   }
 
-  def subscriptionRefiner(onNonMember: RequestHeader => Result = notYetAMemberOn(_)) = new ActionRefiner[AuthRequest, SubReqWithSub] {
-    override def refine[A](request: AuthRequest[A]): SubRequestOrResult[A] = {
-      getSubRequest(request).map(_ toRight onNonMember(request))
+  def subscriptionRefiner(
+      onNonMember: RequestHeader => Result = notYetAMemberOn(_)) =
+    new ActionRefiner[AuthRequest, SubReqWithSub] {
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: weird, indentation goes backwards??

File: frontend/app/actions/ActionRefiners.scala [hunk 3]
```diff
@@ -57,78 +63,119 @@ object ActionRefiners extends LazyLogging {
...
-        new SubscriptionRequest[A](request) with PaidSubscriber {
-          override def subscriber = paid
-        }).toEither
-      )
-  }
+  def paidSubscriptionRefiner(
+      onFreeMember: RequestHeader => Result = notYetAMemberOn(_)) =
+    new ActionRefiner[SubReqWithSub, SubReqWithPaid] {
+      override protected def refine[A](request: SubReqWithSub[A]
+          ): Future[Either[Result, SubReqWithPaid[A]]] =
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: weird

File: frontend/app/actions/ActionRefiners.scala [hunk 4]
```diff
@@ -57,78 +63,119 @@ object ActionRefiners extends LazyLogging {
...
-        }, _ => onPaidMember(request)).swap.toEither // we convert paid to a response, thus giving Free \/ Response
-      )                                              // but we need Either[Response, Free], hence the swap and toEither
-  }
+  def freeSubscriptionRefiner(
+      onPaidMember: RequestHeader => Result = notYetAMemberOn(_)) =
+    new ActionRefiner[SubReqWithSub, SubReqWithFree] {
+      override protected def refine[A](request: SubReqWithSub[A]
+          ): Future[Either[Result, SubReqWithFree[A]]] =
+        Future.successful(request.paidOrFreeSubscriber
+              .bimap(free =>
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: very weird

File: frontend/app/actions/ActionRefiners.scala [hunk 5]
```diff
@@ -57,78 +63,119 @@ object ActionRefiners extends LazyLogging {
...
+                       new SubscriptionRequest[A](request)
+                       with FreeSubscriber {
+                     override def subscriber = free
+                 },
+                     _ => onPaidMember(request))
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: wtf?

File: frontend/app/actions/ActionRefiners.scala [hunk 6]
```diff
@@ -57,78 +63,119 @@ object ActionRefiners extends LazyLogging {
...
-      if (!request.touchpointBackend.memberService.subscriptionUpgradableTo(request.subscriber.subscription, targetTier)) {
-        Future.successful(Some(Ok(views.html.tier.upgrade.unavailable(request.subscriber.subscription.plan.tier, targetTier))))
-      } else {
-        Future.successful(None)
+  def checkTierChangeTo(targetTier: PaidTier) =
+    new ActionFilter[SubReqWithSub] {
+      override protected def filter[A](
+          request: SubReqWithSub[A]): Future[Option[Result]] = {
+        if (!request.touchpointBackend.memberService.subscriptionUpgradableTo(
+                request.subscriber.subscription, targetTier)) {
```
Comment (joelochlann) [inline frontend/app/actions/ActionRefiners.scala]: this is horrible

**Comment by olafurpg** [comment]
Hello! I'm the scalafmt author, thanks for trying it out 😄 I agree that scalafmt produces some horrible output in this diff. I cordially invite you to report issues on the biggest offences and participate in making it better! Don't forget to include your expected output.

Quick comments on this codebase:
* it's clearly not written with an 80 column character limit in mind. A `--maxColumn 120` might yield better results.
* it uses many Scala idioms (for example, default parameter values and [refined types](http://ktoso.github.io/scala-types-of-types/#refined-types-refinements)) that I haven't paid so much attention to. Mostly because I don't use them. Understanding your expectations here is certainly better than me guessing blindly what might look nice.

**Comment by AWare** [comment]
Hey @olafurpg, 

Firstly, thanks for writing scalafmt, it's a daunting task to write an autoformatter for any language. 

I'll have a look at the output we get with a slightly wider maxColumn. For some of these, while scalafmt doesn't give great output, it's not obvious what the correct formatting should be. Also, scalafmt has improved lots since we last tried it. Last time it wasn't able to parse our full codebase, and I had to run it on a few individual files.

**Pull request closed without merging**